test/ariaqueryhandler.spec.ts

```diff
@@ -0,0 +1,526 @@
+        `
+      );
+    });
+
+    it('should find by name', async () => {
+      const { page } = getTestState();
+      const menu = await page.$('aria/menu-label1');
+      const id = await menu.evaluate((b: Element) => b.id);
+      expect(id).toBe('mnu1');
+    });
+
+    it('should find by name', async () => {
+      const { page } = getTestState();
+      const menu = await page.$('aria/menu-label2');
+      const id = await menu.evaluate((b: Element) => b.id);
+      expect(id).toBe('mnu2');
+    });
+
+    it('$$eval should handle many elements (aria)', async () => {
+      const { page } = getTestState();
+      await page.evaluate(
+        `
+        for (var i = 0; i <= 10000; i++) {
+            const button = document.createElement('button');
+            button.textContent = i;
+            document.body.appendChild(button);
+        }
+        `
+      );
+      const sum = await page.$$eval('aria/&button', (buttons) =>
+        buttons.reduce((acc, button) => acc + Number(button.textContent), 0)
+      );
+      expect(sum).toBe(50005000);
+    });
+  });
+
+  describe('waitForSelector (aria)', function () {
+    const addElement = (tag) =>
+      document.body.appendChild(document.createElement(tag));
+
+    it('should immediately resolve promise if node exists', async () => {
+      const { page, server } = getTestState();
+      await page.goto(server.EMPTY_PAGE);
+      await page.evaluate(addElement, 'button');
+      await page.waitForSelector('aria/&button');
+    });
+
+    it('should work independently of `exposeFunction`', async () => {
+      const { page, server } = getTestState();
+      await page.goto(server.EMPTY_PAGE);
+      await page.exposeFunction('ariaQuerySelector', (a, b) => a + b);
+      await page.evaluate(addElement, 'button');
+      await page.waitForSelector('aria/&button');
+      const result = await page.evaluate('globalThis.ariaQuerySelector(2,8)');
+      expect(result).toBe(10);
+    });
+
+    it('should work with removed MutationObserver', async () => {
+      const { page } = getTestState();
+
+      await page.evaluate(() => delete window.MutationObserver);
+      const [handle] = await Promise.all([
+        page.waitForSelector('aria/anything'),
+        page.setContent(`<h1>anything</h1>`),
+      ]);
+      expect(
+        await page.evaluate((x: HTMLElement) => x.textContent, handle)
+      ).toBe('anything');
+    });
+
+    it('should resolve promise when node is added', async () => {
+      const { page, server } = getTestState();
+
+      await page.goto(server.EMPTY_PAGE);
+      const frame = page.mainFrame();
+      const watchdog = frame.waitForSelector('aria/&heading');
+      await frame.evaluate(addElement, 'br');
+      await frame.evaluate(addElement, 'h1');
+      const eHandle = await watchdog;
+      const tagName = await eHandle
+        .getProperty('tagName')
+        .then((e) => e.jsonValue());
```

Comment:
nit: avoid single-letter variable names where possible (I know the old test set a bad example here)

`e` -> `element`?